### ISSUE TITLE
Unable to access ipv6 website

### ISSUE THREAD

**Issue by Trustedinstall** [AdguardTeam/AdguardForAndroid#4198]
[ Yes ] I am running the latest version

AdGuard version:

v3.6.8 (46)
Filtering mode:

Local VPN
Device:

OnePlus 3T
Operating system and version:

Mokee Android 10
Root access:

Yes

AdGuard uses a custom DNS service, which is a DNS service built locally with AdGuardHome, port 5353.
It is not possible to pass the connection test of ipv6 (https://test-ipv6.com/) and it is impossible to access websites with only ipv6 addresses (https://6.ipw.cn).
After turning off the protection, it can be tested by ipv6. In addition, after switching networks, it is also possible to access iPV6 websites normally for a period of time, such as switching from wifi to mobile network or vice versa.
[adguard_logs_3.6.46_2806_121317.zip](https://github.com/AdguardTeam/AdguardForAndroid/files/8997792/adguard_logs_3.6.46_2806_121317.zip)

**Comment by vecinohk** [comment]
I have the same problem on S21 FE with Android 12.

**Comment by maxikuzmin** [comment]
@Trustedinstall @vecinohk  issue in the works

**Comment by maxikuzmin** [comment]
@Trustedinstall @vecinohk which DNS protocol are you using? 
Also take a screenshot from the "Query log" in AdGuard Home, you can paste it directly here. 
Can you please tell me if you have the Settings -> DNS settings -> Disable resolving of IPV6 addresses checked?

**Comment by Trustedinstall** [comment]
> @Trustedinstall @vecinohk您使用的是哪个 DNS 协议？ 也可以从 AdGuard Home 中的“查询日志”截图，你可以直接粘贴到这里。 你能告诉我你是否勾选了设置 -> DNS 设置 -> 禁用解析 IPV6 地址？

AdGuard Home uses only DoH to query upstream dns
The query log only receives query requests for A records, not AAAA records.
I didn't check Disable resolving of IPV6 addresses checked

**Comment by Trustedinstall** [comment]
> @Trustedinstall @vecinohk which DNS protocol are you using? Also take a screenshot from the "Query log" in AdGuard Home, you can paste it directly here. Can you please tell me if you have the Settings -> DNS settings -> Disable resolving of IPV6 addresses checked?

AdGuard Home uses only DoH to query upstream dns
The query log only receives query requests for A records, not AAAA records.
I didn't check Disable resolving of IPV6 addresses checked

**Comment by maxikuzmin** [comment]
@Trustedinstall @vecinohk does it work on sites without DNS? Does filtering work with DNS? Какое значение стоит в low level settings -> pref.ipv6.routes.excluded?

**Comment by Trustedinstall** [comment]
> @Trustedinstall @vecinohk does it work on sites without DNS? Does filtering work with DNS? What's the value in low level settings -> pref.ipv6.routes.excluded?

There is no DNS filter enabled, the value of pref.ipv6.routes.excluded is empty, I can't even ping the ipv6 address, it feels like ipv6 is disabled, but pref.vpn.ipv6.disable is disabled.
If I set AdGuard's dns as the IP of a computer running AdGuard Home on a local area network, that computer can receive query requests for AAAA records, but still can't ping the ipv6 address. After turning off AdGuard, reconnecting to wifi will be able to access the ipv6 address normally.

**Comment by maxikuzmin** [comment]
@Trustedinstall we're still looking into the problem. Please wait

**Comment by maxikuzmin** [comment]
@Trustedinstall Is the issue still relevant? I have now tested on the latest version 3.6.10 and I am adding DNS with IPv6

**Comment by Trustedinstall** [comment]
> @Trustedinstall Is the issue still relevant? I have now tested on the latest version 3.6.10 and I am adding DNS with IPv6

Yes, still relevant

**Comment by Trustedinstall** [comment]
> @Trustedinstall Is the issue still relevant? I have now tested on the latest version 3.6.10 and I am adding DNS with IPv6

Yes, it is still relevant. in version 3.6.10 there is still the problem of not being able to ping ipv6 addresses. it is still the same as before. it works fine for the first few minutes with adguard on and then the ipv6 network is not accessible.

**Comment by maxikuzmin** [comment]
@Trustedinstall it turns out that in my case I changed my network and I started using DNS with IPv6 added. The thing is that some networks / operators do not have the function "IPv6". Try to change your operator / provider

**Comment by Trustedinstall** [comment]
> @Trustedinstall it turns out that in my case I changed my network and I started using DNS with IPv6 added. The thing is that some networks / operators do not have the function "IPv6". Try to change your operator / provider

I don't think it's a carrier issue because everything works fine when I turn off Adguard and ipv6test passes the test with a perfect score. the same goes for the data network.

**Comment by Trustedinstall** [comment]
> @Trustedinstall事实证明，在我的情况下，我改变了我的网络，并开始使用添加了 IPv6 的 DNS。问题是某些网络/运营商没有“IPv6”功能。尝试更改您的运营商/提供商

Maybe I should change the title of the problem to Can't ping ipv6 addresses with AdGuard enabled

**Comment by Trustedinstall** [comment]
> > @Trustedinstall it turns out that in my case I changed my network and I started using DNS with IPv6 added. The thing is that some networks / operators do not have the function "IPv6". Try to change your operator / provider
> 
> I don't think it's a carrier issue because everything works fine when I turn off Adguard and ipv6test passes the test with a perfect score. the same goes for the data network.

Maybe I should change the title of the problem to Can't ping ipv6 addresses with AdGuard enabled
